src/modules/domains/domains.service.ts

```diff
@@ -75,28 +85,48 @@ export class DomainsService {
       ensRegistryAddress,
       provider,
     });
-    ensPublicResolverAddress &&
+    this._ensRegistryAddress = ensRegistryAddress;
+    if (chainId === EWC_CHAIN_ID) {
```

Comment:
Hmm, I think we should try to avoid this special conditional for EWC chain. Would the changes from https://github.com/energywebfoundation/iam-contracts/pull/113 help with avoiding this?